### PR TITLE
feat(docs): Add testing related remarks to user setup page

### PR DESCRIPTION
**Pull request by caksoylar**
I added a few improvements to the end of user setup cautioning against common issues that people run into when they are first testing their keyboard. These are related to: (a) encouraging testing over USB before BT, (b) explaining peripherals shouldn't be tested before central for split keyboards, (c) linking to troubleshooting for pairing two halves in case of issues, (d) emphasizing BT profiles' importance a little more.

### CODE REVIEW

File: docs/docs/user-setup.md [hunk 1]
```diff
@@ -193,14 +193,31 @@ To flash the firmware, first put your board into bootloader mode by double click
...
+over bluetooth. Peripheral half will only send keystrokes to the central once they are paired and connected. For this reason it is
+recommended to test the left half of a split keyboard first.
+
+:::
 
 ## Wirelessly Connecting Your Keyboard
 
-ZMK will automatically advertise itself as connectable if it is not currently connected to a device. You should be able to see your keyboard from the bluetooth scanning view of your laptop or phone / tablet. It is reported by some users that the connections with Android / iOS devices are generally smoother than with laptops, so if you have trouble connecting, you could try to connect from your phone or tablet first to eliminate any potential hardware issues.
+ZMK will automatically advertise itself as connectable if it is not currently connected to a device. You should be able to see your keyboard from the bluetooth scanning view of your computer or phone / tablet. It is reported by some users that the connections with Android / iOS devices are generally smoother than with laptops, so if you have trouble connecting, you could try to connect from your phone or tablet first to eliminate any potential hardware issues with bluetooth receivers.
```
Comment (Hello9999901) [inline docs/docs/user-setup.md:209]: Really detailed stuff! Perhaps "your computer or phone / tablet" can be shortened to just "your device"?

**Review by Hello9999901 — CHANGES_REQUESTED** [review]

File: docs/docs/user-setup.md (hunk 1 above)
Comment (caksoylar) [reply docs/docs/user-setup.md:209]: We could probably replace it here but this is the most beginner-friendly page of the docs, so maybe it doesn't hurt to be a bit specific/give examples.

In general I am scared people might think "device" refers to the keyboard, though probably not in this specific spot :)

**Review by Hello9999901 — APPROVED** [review]

**Review by BenRoe — APPROVED** [review]
Useful and beginner friendly explanation.
Looks good to me.

**Comment by caksoylar** [comment]
Anyone else have any concerns or improvement suggestions?

**Review by petejohanson — APPROVED** [review]
LGTM!

**Pull request merged**